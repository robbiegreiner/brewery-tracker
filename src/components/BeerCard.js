import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BeerCard = ({ beer, setCurrentBeer, writeUserData, user }) => {
  return (
    <div className='beer-card' >
      <h2 onClick={() => setCurrentBeer(beer)}>{beer.name}</h2>
      <h3>{beer.brewies ? beer.breweries[0].name : 'no brewery info'}</h3>
      <h3>{beer.style ? beer.style.name : 'no style'}</h3>
      <h3>{beer.abv}</h3>
      <Link to={'/beer/' + beer.id}><h4>View Beer Details</h4></Link>
      <h4 onClick={() => writeUserData(user.id, 'beer', beer.id )}>FAVORITE</h4>
    </div>
  );
};

BeerCard.propTypes = {
  beer: PropTypes.object,
  setCurrentBeer: PropTypes.func,
  writeUserData: PropTypes.func,
  user: PropTypes.object
};

export default BeerCard;
