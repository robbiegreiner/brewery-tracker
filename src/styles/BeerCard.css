.beer-card {
  width: 200px;
  height: 300px;
  border: 2px solid black;
  margin: 15px 15px 15px 15px; }
