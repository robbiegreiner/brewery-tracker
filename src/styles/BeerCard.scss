.beer-card, .favorite-card {
  width: 300px;
  height: 510px;
  border: 3px solid black;
  border-radius: 10px;
  margin: 15px 15px 15px 15px;
  background-color: #2C4154;
  color: white;
  letter-spacing: 1px;
}

.beer-card a, .favorite-card a {
  color: white;
}

.beer-card h2, .favorite-card h2 {
  margin-top: 20px;
  margin-bottom: 10px;

}

.beer-logo, .brewery-logo {
  width: 300px;
  height: 250px;
  margin: 0;
  filter: opacity(.7);
}

.brewery-logo {
  height: 220px;
  margin-top: 20px;
  margin-bottom: 10px;
}

.beer-icon, .brewery-icon {
  width: 200px;
  margin-top: 30px;
  margin-bottom: 20px;
}

.brewery-icon{
  width: 150px;
}

.favorite-card {
  border: 3px solid #FF644E;
}

.style, .abv {
  margin-bottom: 10px;
}

.view-beer {
  margin-bottom: 0px;
  margin-top: 0px;
  margin-right: 15px;
}

.bottom-btns{
  display: flex;
  width: 100%;
  justify-content: center;
  margin-top: 20px;
  font-size: 20px;
}

.favorite {
  cursor: pointer;
}

.website {
  font-size: 12px;
  margin-bottom: 20px;
  color: white;
}

.type {
  margin-top: 10px;
}

.description {
  width: 100px;
  height: 100px;
  white-space: nowrap;
  overflow: hidden;
}
