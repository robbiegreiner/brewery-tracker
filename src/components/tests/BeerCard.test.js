import React from "react";
import BeerCard from '../BeerCard';
import { shallow } from "enzyme";

describe('BeerCard component', () => {
  const mockFn = jest.fn();

  const match = {
  isExact: true,
  params: {
    name: 'the-topic-name'
  },
  path: '/topics/:name',
  url: '/topics/the-topic-name'
};

  const wrapper = shallow(<BeerCard
    user={{}}
    writeUserData={mockFn}
    setCurrentBeer={mockFn}
    beer={{}}
    match={{  params: { beer_id: 'vcxy7z' }  }}
    key={'vyzgyz'}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
