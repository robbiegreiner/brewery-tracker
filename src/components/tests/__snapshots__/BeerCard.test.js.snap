// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerCard
    beer={Object {}}
    match={
        Object {
            "params": Object {
              "beer_id": "vcxy7z",
            },
          }
    }
    setCurrentBeer={[Function]}
    user={Object {}}
    writeUserData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          onClick={[Function]}
/>,
        <h3>
          no brewery info
</h3>,
        <h3>
          no style
</h3>,
        <h3 />,
        <Link
          replace={false}
          to="/beer/undefined"
>
          <h4>
                    View Beer Details
          </h4>
</Link>,
        <h4
          onClick={[Function]}
>
          FAVORITE
</h4>,
      ],
      "className": "beer-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "no brewery info",
        },
        "ref": null,
        "rendered": "no brewery info",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "no style",
        },
        "ref": null,
        "rendered": "no style",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h4>
            View Beer Details
</h4>,
          "replace": false,
          "to": "/beer/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "View Beer Details",
          },
          "ref": null,
          "rendered": "View Beer Details",
          "type": "h4",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "FAVORITE",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "FAVORITE",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            onClick={[Function]}
/>,
          <h3>
            no brewery info
</h3>,
          <h3>
            no style
</h3>,
          <h3 />,
          <Link
            replace={false}
            to="/beer/undefined"
>
            <h4>
                        View Beer Details
            </h4>
</Link>,
          <h4
            onClick={[Function]}
>
            FAVORITE
</h4>,
        ],
        "className": "beer-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "no brewery info",
          },
          "ref": null,
          "rendered": "no brewery info",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "no style",
          },
          "ref": null,
          "rendered": "no style",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h4>
              View Beer Details
</h4>,
            "replace": false,
            "to": "/beer/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "View Beer Details",
            },
            "ref": null,
            "rendered": "View Beer Details",
            "type": "h4",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "FAVORITE",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "FAVORITE",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
